src/workerd/api/global-scope.h

```diff
@@ -756,9 +756,8 @@ class ServiceWorkerGlobalScope: public WorkerGlobalScope {
     JSG_NESTED_TYPE(HTMLRewriter);
 
     // Performance API
-    JSG_NESTED_TYPE(Performance);
-
     if (flags.getEnableGlobalPerformanceClasses() || flags.getEnableNodeJsPerfHooksModule()) {
+      JSG_NESTED_TYPE(Performance);
```

Comment:
this may not be enough... let's add a regression test to ensure that there aren't other hidden gotchas